pkg/core/mpt/compat_test.go

```diff
@@ -0,0 +1,360 @@
+package mpt
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func prepareMPTCompat() *Trie {
+	b := NewBranchNode()
+	r := NewExtensionNode([]byte{0x0a, 0x0c}, b)
+	v1 := NewLeafNode([]byte{0xab, 0xcd}) //key=ac01
+	v2 := NewLeafNode([]byte{0x22, 0x22}) //key=ac
+	v3 := NewLeafNode([]byte("existing")) //key=acae
+	v4 := NewLeafNode([]byte("missing"))
+	h3 := NewHashNode(v3.Hash())
+	e1 := NewExtensionNode([]byte{0x01}, v1)
+	e3 := NewExtensionNode([]byte{0x0e}, h3)
+	e4 := NewExtensionNode([]byte{0x01}, v4)
+	b.Children[0] = e1
+	b.Children[10] = e3
+	b.Children[16] = v2
+	b.Children[15] = NewHashNode(e4.Hash())
+
+	tr := NewTrie(r, true, newTestStore())
+	tr.putToStore(r)
+	tr.putToStore(b)
+	tr.putToStore(e1)
+	tr.putToStore(e3)
+	tr.putToStore(v1)
+	tr.putToStore(v2)
+	tr.putToStore(v3)
+
+	return tr
+}
+
+// TestCompatibility contains tests present in C# implementation.
+// https://github.com/neo-project/neo-modules/blob/master/tests/Neo.Plugins.StateService.Tests/MPT/UT_MPTTrie.cs
+// There are some differences, though:
+// 1. In our implementation delete is silent, i.e. we do not return an error is the key is missing.
+//    However, we do return error when contents of hash node are missing from the store
+//    (corresponds to exception in C# implementation).
+// 2. If `GetProof` key is missing from the trie, we return error, while C# node just returns empty proof
+//    with no exception.
```

Comment:
Empty as no proof or path until the last node as proof of non-existence?

Reply:
`null` as no proof